File: sdkwork-ai-sdk-flutter/lib/src/api/music2.dart
import '../http/client.dart';
import '../models.dart';

class Music2Api {
  final HttpClient _client;
  
  Music2Api(this._client);

  /// Generate music
  Future<SunoMusic?> generate(MusicGenerationRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/music/generations'), body: body, contentType: 'application/json');
    return response is SunoMusic ? response : null;
  }

  /// Retrieve music
  Future<SunoMusic?> retrieve(String musicId) async {
    final response = await _client.get(ApiPaths.aiPath('/music/${musicId}'));
    return response is SunoMusic ? response : null;
  }

  /// Delete music
  Future<SunoMusicDeleteResponse?> deleteMusic(String musicId) async {
    final response = await _client.delete(ApiPaths.aiPath('/music/${musicId}'));
    return response is SunoMusicDeleteResponse ? response : null;
  }

  /// List music
  Future<SunoMusicList?> listMusic(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/music'), params: params);
    return response is SunoMusicList ? response : null;
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/music2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import MusicGenerationRequest, SunoMusic, SunoMusicDeleteResponse, SunoMusicList

class Music2Api:
    """music API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def generate(self, body: MusicGenerationRequest) -> SunoMusic:
        """Generate music"""
        return self._client.post(f"/ai/v3/music/generations", json=body)

    def retrieve(self, musicId: str) -> SunoMusic:
        """Retrieve music"""
        return self._client.get(f"/ai/v3/music/{musicId}")

    def delete_music(self, musicId: str) -> SunoMusicDeleteResponse:
        """Delete music"""
        return self._client.delete(f"/ai/v3/music/{musicId}")

    def list_music(self, params: Optional[Dict[str, Any]] = None) -> SunoMusicList:
        """List music"""
        return self._client.get(f"/ai/v3/music", params=params)
